Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -405,6 +405,23 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: ArrayTypeSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if node.leftSquareBracket.presence == .missing {
+      addDiagnostic(
+        node.leftSquareBracket,
```

Comment:
Can you anchor the diagnostic at `node.rightSquareBracket`. Since it’s talking about the unexpected `]`, I think it should also point to the `]`.